weave/integrations/dspy/dspy.py

```diff
@@ -0,0 +1,189 @@
+import importlib
```

Comment:
Requested changes: We need to name the ops in such a way that they are properly namespaced, at a minimum, 
1. all names should start with `MODULE.`
2. Please don't use `__call__` for the names - instead just exclude that path component

Reply:
I tried to do this using `weave.op(display_name="dspy.<fn-name>")`, it didn't work. Can you please suggest how we can do this?

Reply:
please use `name`

Reply:
`display_name` is confusing and shouldn't be here

Reply:
that is on us - we are refactoring this

Reply:
oh
, you might need:
```
        op = weave.op()(fn)
        op.name = name  # type: ignore
```

this is how i did it in litellm

Reply:
Hi @tssweeney,
I tried your recommendation, but it didn't seem to work.

Here's a simple PoC implementation I made of the solution you recommended: https://github.com/wandb/weave/pull/1789

I'm testing the changes with the following snippet:
```python
import dspy
import weave

weave.init(project_name="test-project")

gpt3_turbo = dspy.OpenAI(model="gpt-3.5-turbo-1106", max_tokens=300)
dspy.configure(lm=gpt3_turbo)
prediction = gpt3_turbo("hello! this is a raw prompt to GPT-3.5")
```

Result:
<img width="1440" alt="image" src="https://github.com/wandb/weave/assets/19887676/946828e1-b03b-4d30-8c64-4882021ad3a2">

Can you please recommend if I'm doing anything wrong or not?